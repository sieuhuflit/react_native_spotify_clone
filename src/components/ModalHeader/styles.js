import { StyleSheet } from 'react-native';
import { colors, device } from '../../constants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 24,
    paddingTop: device.iPhoneX ? 48 : 24,
  },
  containerText: {
    alignItems: 'center',
    flex: 5,
    justifyContent: 'center',
  },
  text: {
    fontSize: 16,
    color: colors.white,
    textAlign: 'center',
  },
  left: {
    alignItems: 'flex-start',
    flex: 1,
    justifyContent: 'center',
  },
  right: {
    alignItems: 'flex-end',
    flex: 1,
    justifyContent: 'center',
  },
});

export default styles;
