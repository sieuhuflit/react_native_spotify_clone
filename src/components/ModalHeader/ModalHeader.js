import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import TouchIcon from '../TouchIcon';
import styles from './styles';

const ModalHeader = ({ left, leftPress, right, rightPress, style, text }) => (
  <View style={[styles.container, style]}>
    {left && <TouchIcon icon={left} onPress={leftPress} style={styles.left} />}
    {!left && <View style={styles.left} />}
    {text && (
      <View style={styles.containerText}>
        <Text style={styles.text}>{text}</Text>
      </View>
    )}
    {right && (
      <TouchIcon icon={right} onPress={rightPress} style={styles.right} />
    )}
    {!right && <View style={styles.right} />}
  </View>
);

ModalHeader.defaultProps = {
  left: null,
  leftPress: () => null,
  right: null,
  rightPress: () => null,
  style: {},
  text: null
};

ModalHeader.propTypes = {
  left: PropTypes.element,
  leftPress: PropTypes.func,
  right: PropTypes.element,
  rightPress: PropTypes.func,
  style: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.number,
    PropTypes.object
  ]),
  text: PropTypes.string
};

export default ModalHeader;
