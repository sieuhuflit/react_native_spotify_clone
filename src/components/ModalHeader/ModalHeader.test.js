import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import ModalHeader from './ModalHeader';

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides);
  const wrapper = shallow(<ModalHeader {...props} />);
  return {
    props,
    wrapper
  };
};

describe('ModalHeader check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ModalHeader check render correctly', () => {
  test('should render 1 Text when has text props ', () => {
    const { wrapper } = setup({
      text: 'Hi'
    });
    expect(wrapper.find('Text')).toHaveLength(1);
  });

  test('should has 2 TouchIcon when has left and right props ', () => {
    const { wrapper } = setup({
      left: <View />,
      right: <View />
    });
    expect(wrapper.find('TouchIcon')).toHaveLength(2);
  });

  test('should render 2 View when left and right props is null ', () => {
    const { wrapper } = setup({
      left: null,
      right: null
    });
    expect(
      wrapper
        .find('View')
        .at(0)
        .children()
    ).toHaveLength(2);
  });
});

describe('ModalHeader check simulate click', () => {
  test('should leftPress and rightPress work', () => {
    const { wrapper, props } = setup({
      left: <View />,
      right: <View />,
      leftPress: jest.fn(),
      rightPress: jest.fn()
    });
    wrapper
      .find('TouchIcon')
      .at(0)
      .props()
      .onPress();
    expect(props.leftPress).toBeCalledTimes(1);
    wrapper
      .find('TouchIcon')
      .at(1)
      .props()
      .onPress();
    expect(props.rightPress).toBeCalledTimes(1);
  });
});
