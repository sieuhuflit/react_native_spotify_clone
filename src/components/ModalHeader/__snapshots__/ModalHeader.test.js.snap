// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader check snapshot should match snapshot 1`] = `ShallowWrapper {}`;
